phase2/docker-images/user-data.yaml

```diff
@@ -0,0 +1,141 @@
+      uyMKH5NMVy9c+A9yqq9NF6M6d6/ehKfBFFLG9BX+XLBATvf1ZemGVHQusCQebTGv
+      0C0V9yqtdPdRWVIEhHxyNHATaVYOafTj/EF0lDxLl6zDT6trRV5n9F1VCEh4Aal8
+      L5MxVPcIZVO7NHT2EkQgn8CvWjV3oKl2GopZF8V4XdJRl90U/WDv/6cmfI08GkzD
+      YBHhS8ULWRFwGKobsSTyIvnbk4NtKdnTGyTJCQ8+6i52s+C54PiNgfj2ieNn6oOR
+      7d+bNCcG1CdOYY+ZXVOcsjl73UYvtJrO0Rl/NpYERkZ5d/tzw4jZ6FCXgggA/Zxc
+      jk6Y1ZvIm8Mt8wLRFH9Nww+FVsCtaCXJLP8DlJLASMD9rl5QS9Ku3u7ZNrr5HWXP
+      HXITX660jglyshch6CWeiUATqjIAzkEQom/kEnOrvJAtkypRJ59vYQOedZ1sFVEL
+      MXg2UCkD/FwojfnVtjzYaTCeGwFQeqzHmM241iuOmBYPeyTY5veF49aBJA1gEJOQ
+      TvBR8Q==
+      =Fm3p
+      -----END PGP PUBLIC KEY BLOCK-----
+
+packages:
+  - docker-engine
+
+write_files:
+
+  - path: /etc/kubernetes-anywhere/base.env
+    permissions: '0644'
+    owner: root
+    content: |
+      WEAVE_VERSION="1.5.2"
+      WEAVE_SCRIPT_VERSION="v1.5.2"
+      CLOUD_PROVIDER="aws" ## can probably auto-detect it
+      DOCKER_ARGS="--host=fd:// --storage-driver=overlay"
+
+  - path: /etc/kubernetes-anywhere/images.env
+    permissions: '0644'
+    owner: root
+    content: |
+      KUBERNETES_ANYWHERE_TOOLBOX_IMAGE="weaveworks/kubernetes-anywhere:toolbox-v1.2-<__TODO__>"
+      #KUBERNETES_ANYWHERE_APISERVER_IMAGE="weaveworks/kubernetes-anywhere:apiserver-v1.2"
+      #KUBERNETES_ANYWHERE_SCHEDULER_IMAGE="weaveworks/kubernetes-anywhere:scheduler-v1.2"
+      #KUBERNETES_ANYWHERE_CONTROLLER_MANAGER_IMAGE="weaveworks/kubernetes-anywhere:controller-manager-v1.2"
+      #KUBERNETES_ANYWHERE_KUBELET_IMAGE="weaveworks/kubernetes-anywhere:kubelet-v1.2"
+      #KUBERNETES_ANYWHERE_PROXY_IMAGE="weaveworks/kubernetes-anywhere:proxy-v1.2"
+
+  - path: /etc/systemd/system/docker.socket
+    permissions: '0644'
+    owner: root
+    content: |
+      [Unit]
+      Description=Docker Socket for the API
+      PartOf=docker.service
+      [Socket]
+      ListenStream=/var/run/docker.sock
+      SocketMode=0660
+      [Install]
+      WantedBy=sockets.target
+
+  - path: /etc/systemd/system/docker.service
+    permissions: '0644'
+    owner: root
+    content: |
+      [Unit]
+      Description=Docker Application Container Engine
+      Documentation=http://docs.docker.com
+      After=docker.socket network.target
+      Requires=docker.socket
+      [Service]
+      EnvironmentFile=/etc/kubernetes-anywhere/base.env
+      EnvironmentFile=-/etc/kubernetes-anywhere/local.env
+      ExecStart=/usr/bin/docker daemon $DOCKER_ARGS
+      Restart=on-failure
+      LimitNOFILE=1048576
+      LimitNPROC=1048576
+      [Install]
+      WantedBy=weave-network.target
+      WantedBy=weave.service
+
+  - path: /tmp/kubernetes-anywhere-bootstrap.sh
+    owner: root
+    permissions: '0755'
+    content: |
+      #!/bin/bash -x
+
+      set -o errexit
+      set -o pipefail
+      set -o nounset
+
+      systemctl daemon-reload
+
+      docker -v || env DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated --force-yes  docker-engine
```

Comment:
This a workaround for https://github.com/docker/docker/issues/22599, which is a bit obscure, but I have managed to hit it at least once.